### PR TITLE
Use Anonymous Struct for sret

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: src/compiler.jl [hunk 1]
```diff
@@ -4746,15 +4746,15 @@ end
         else
             return quote
                 Base.@_inline_meta
-                sret = Box{$(Tuple{sret_types...})}()
+                sret = Box{$(AnonymousStruct(Tuple{sret_types...}))}()
```
Comment (wsmoses) [inline src/compiler.jl:4749]: Can we just have this subsume the prior special casing of if it contains an any or not allocated inline as well?

Comment (vchuravy) [reply src/compiler.jl:4749]: Probably, want to make that change?

**Comment by wsmoses** [comment]
```
Test Summary: | Pass  Total
Bessel        |   24     24
AbstractType calling convention: Test Failed at /mnt/Data/git/Enzyme.jl/test/runtests.jl:431
  Expression: Enzyme.autodiff(dxdt_pred, Active(1.0))
    Expected: UndefRefError
  No exception thrown
Stacktrace:
 [1] macro expansion
   @ /mnt/Data/git/Enzyme.jl/test/runtests.jl:431 [inlined]
 [2] macro expansion
   @ /mnt/Data/git/Enzyme.jl/julia-1.7.2/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]
 [3] top-level scope
   @ /mnt/Data/git/Enzyme.jl/test/runtests.jl:429
Test Summary:                   | Fail  Total
AbstractType calling convention |    1      1
ERROR: LoadError: Some tests did not pass: 0 passed, 1 failed, 0 errored, 0 broken.
in expression starting at /mnt/Data/git/Enzyme.jl/test/runtests.jl:428
ERROR: Package Enzyme errored during testing
```

Seemignly not sufficient, throwing this back to you @vchuravy

**Comment by wsmoses** [comment]
Wait actually, that is good not to have thrown, one second.

**Comment by vchuravy** [comment]
I plan to merge this as soon as CI agrees, it's a strict improvement and the remaining issues with sret can be fixed independently.

**Comment by wsmoses** [comment]
Check also https://github.com/EnzymeAD/Enzyme.jl/issues/293 which was a minimal example not resolved by iirc.

**Comment by wsmoses** [comment]
(it also is referenced by several other errors that became variants of that issue, some of which were in fact resolved by the PR)

**Pull request merged**